etl.py

```diff
@@ -49,55 +56,115 @@ def chunks(full_list, chunk_size, start_chunk=1, end_chunk=None):
 
 
 class UploadJob(object):
-    def __init__(self, csv_file_path, x_column="longitude", y_column="latitude", srid=4326):
+    def __init__(self, csv_file_path, x_column="longitude",
+                 y_column="latitude", srid=4326, file_encoding=FILE_ENCODING,
+                 force_no_geometry=FORCE_NO_GEOMETRY):
         self.csv_file_path = csv_file_path
         self.x_column = x_column
         self.y_column = y_column
         self.srid = srid
+        self.file_encoding = file_encoding
+        self.force_no_geometry = force_no_geometry
 
     def run(self):
         raise NotImplemented
 
     def regenerate_overviews(self):
-        query = 'select CDB_CreateOverviews(\'{table}\'::regclass)'.format(table=CARTO_TABLE_NAME)
+        query = 'select CDB_CreateOverviews(\'{table}\'::regclass)'.\
+            format(table=CARTO_TABLE_NAME)
         job_result = bsql.create(query)
         return job_result['job_id']
 
     def check_job(self, job_id):
         return bsql.read(job_id)
 
+    def create_geom_query(self, record):
+        null_result = NULL_VALUE + ","
+        if self.force_no_geometry:
+            return null_result
+
+        longitude = self.get_longitude(record)
+        latitude = self.get_latitude(record)
+
+        if longitude is None or latitude is None:
```

Comment:
what if `longitude` or `latitude` are `DEFAULT_COORD`? is that also something to consider?

Reply:
yep, adding the condition as well